### ISSUE TITLE
Improve queuing and task routing for repo syncing

### ISSUE THREAD

**Issue by katia-sentry** [codecov/engineering-team#7]
The goal of this ticket is to improve the queuing and task routing approach for repo syncing tasks to prioritize and execute these tasks sooner. This should lead to less wait time for the users to improve their overall experience with repo syncing tasks.

**What to do**

Investigate current queuing and task routing approach and identify any inefficiencies or bottlenecks

Propose improvements that will result in repo syncing tasks being prioritized and executed more quickly

Implement the chosen approach, or create ticket(s) if it’s more appropriate to do so

**Acceptance Criteria**

- An investigation on current queuing and task routing approach is complete, and inefficiencies are identified
- Improvements are proposed for their potential to ensure repo syncing tasks being prioritized and executed more quickly
- The chosen improvements are implemented (OR, if necessary, separate ticket(s) are created to do this work)
- The changes are thoroughly tested to confirm improved queuing and task routing
- Relevant documentation is updated

jira: https://codecovio.atlassian.net/browse/CODE-3321

This depends on #6
